h5py/_hl/dataset.py

```diff
@@ -242,6 +242,86 @@ def __exit__(self, *args):
             # pylint: disable=protected-access
             self._dset._dxpl.set_dxpl_mpio(h5fd.MPIO_INDEPENDENT)
 
+class ChunkIterator(object):
+    """
+    Class to iterate through list of chunks of a given dataset
+    # Note: Only works with PY3 since as is doesn't have a "next = __next__" redef
```

Comment:
We can lose this note, as h5py 3.0 is going to require Python 3 anyway (but thanks for making it clear!)